examples/tv-casting-app/tv-casting-common/core/CastingPlayer.h

```diff
@@ -119,22 +119,39 @@ class CastingPlayer : public std::enable_shared_from_this<CastingPlayer>
     /**
      * @brief Verifies that a connection exists with this CastingPlayer, or triggers a new session
      * request. If the CastingApp does not have the nodeId and fabricIndex of this CastingPlayer cached on disk,
-     * this will execute the user directed commissioning process.
+     * this will execute the User Directed Commissioning (UDC) process.
      *
-     * @param onCompleted for success - called back with CHIP_NO_ERROR and CastingPlayer *.
-     * For failure - called back with an error and nullptr.
+     * @param connectionCallbacks contains the ConnectCallback and CommissionerDeclarationCallback (Optional).
      * @param commissioningWindowTimeoutSec (Optional) time (in sec) to keep the commissioning window open, if commissioning is
      * required. Needs to be >= kCommissioningWindowTimeoutSec.
      * @param idOptions (Optional) Parameters in the IdentificationDeclaration message sent by the Commissionee to the Commissioner.
      * These parameters specify the information relating to the requested commissioning session.
      * Furthermore, attributes (such as VendorId) describe the TargetApp that the client wants to interact with after commissioning.
-     * If this value is passed in, VerifyOrEstablishConnection() will force User Directed Commissioning, in case the desired
+     * If this value is passed in, VerifyOrEstablishConnection() will force UDC, in case the desired
      * TargetApp is not found in the on-device CastingStore.
      */
-    void VerifyOrEstablishConnection(ConnectCallback onCompleted,
+    void VerifyOrEstablishConnection(ConnectionCallbacks connectionCallbacks,
                                      unsigned long long int commissioningWindowTimeoutSec = kCommissioningWindowTimeoutSec,
                                      IdentificationDeclarationOptions idOptions           = IdentificationDeclarationOptions());
 
+    /**
+     * @brief Continues the UDC process during the Commissioner-Generated passcode commissioning flow by sending a second
+     * IdentificationDeclaration to Commissioner containing CommissionerPasscode and CommissionerPasscodeReady set to true. At this
+     * point it is assumed that the following have occurred:
+     * 1. Client has handled the Commissioner's CommissionerDecelration message with PasscodeDialogDisplayed and
+     * CommissionerPasscode set to true.
+     * 2. Client prompted user to input Passcode from Commissioner.
+     * 3. Client has updated the commissioning session's PAKE verifier using the user input Passcode by updating the CastingApps
+     * CommissionableDataProvider
+     * (matter::casting::core::CastingApp::GetInstance()->UpdateCommissionableDataProvider(CommissionableDataProvider)).
+     *
+     * @param connectionCallbacks contains the ConnectCallback and CommissionerDeclarationCallback (Optional).
+     * @param commissioningWindowTimeoutSec (Optional) time (in sec) to keep the commissioning window open, if commissioning is
```

Comment:
who is the commissioningWindowTimeoutSec  marked as optional? It is not std::optional or chip::Optional. Is there some specific optional value? It just seems to have a default

Reply:
It is optional because it has a default value if not passed in; like in VerifyOrEstablishConnection() above: 
unsigned long long int commissioningWindowTimeoutSec = kCommissioningWindowTimeoutSec,

defined in CastingPlayer.h as:
const unsigned long long int kCommissioningWindowTimeoutSec = 3 * 60; // 3 minutes
Going to figure out why sharadb-amazon made it so big.